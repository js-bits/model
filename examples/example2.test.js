// eslint-disable-next-line import/no-extraneous-dependencies
import { jest } from '@jest/globals';

describe('Examples', () => {
  beforeEach(() => {
    global.console = { log: jest.fn() };
  });
  afterEach(() => {
    jest.resetModules();
  });

  test('Example 2', done => {
    import('./example2.js').then(() => {
      expect(console.log).toHaveBeenCalledTimes(10);
      expect(console.log.mock.calls[0]).toEqual(['Invalid data']);
      expect(console.log.mock.calls[1]).toEqual([{ balance: 'must be a number' }]);
      expect(console.log.mock.calls[2]).toEqual(['Invalid data']);
      expect(console.log.mock.calls[3]).toEqual([{ title: 'required property is not defined' }]);
      expect(console.log.mock.calls[4]).toEqual(['Invalid data']);
      expect(console.log.mock.calls[5]).toEqual([
        {
          account: 'invalid model type',
          category: 'must be a model',
          date: 'required property is not defined',
        },
      ]);
      expect(console.log.mock.calls[6]).toEqual([
        JSON.stringify(
          [
            {
              title: 'Checking Account',
              balance: 123.56,
              limit: null,
            },
            {
              title: 'Credit Card',
              balance: -876.32,
              limit: 1000,
            },
          ],
          null,
          '  '
        ),
      ]);
      expect(console.log.mock.calls[7]).toEqual([
        JSON.stringify(
          {
            parent: null,
            exclude: null,
            title: 'Food',
          },
          null,
          '  '
        ),
      ]);
      expect(console.log.mock.calls[8]).toEqual([new Date('10/09/2010'), 'Dining', 'Checking Account']);
      expect(console.log.mock.calls[9]).toEqual([123.56]);
      done();
    });
  });
});
