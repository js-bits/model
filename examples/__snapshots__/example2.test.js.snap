// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Examples Example 2 1`] = `
Array [
  Array [
    "Data is invalid",
  ],
  Array [
    Array [
      "\\"balance\\": must be a number",
    ],
  ],
  Array [
    "Data is invalid",
  ],
  Array [
    Array [
      "\\"title\\": required property is not defined",
    ],
  ],
  Array [
    "Data is invalid",
  ],
  Array [
    Array [
      "\\"account\\": invalid model type",
      "\\"category\\": invalid model type",
      "\\"date\\": required property is not defined",
    ],
  ],
  Array [
    "[
  {
    \\"title\\": \\"Checking Account\\",
    \\"balance\\": 123.56,
    \\"limit\\": null
  },
  {
    \\"title\\": \\"Credit Card\\",
    \\"balance\\": -876.32,
    \\"limit\\": 1000
  }
]",
  ],
  Array [
    "{
  \\"title\\": \\"Food\\",
  \\"parent\\": null,
  \\"exclude\\": null
}",
  ],
  Array [
    2010-10-09T04:00:00.000Z,
    "Dining",
    "Checking Account",
  ],
  Array [
    123.56,
  ],
]
`;
